receiver/nsxtreceiver/internal/metadata/generated_metrics_v2.go

```diff
@@ -0,0 +1,710 @@
+		return "errored"
+	case AttributePacketTypeSuccess:
+		return "success"
+	}
+	return ""
+}
+
+// MapAttributePacketType is a helper map of string to AttributePacketType attribute value.
+var MapAttributePacketType = map[string]AttributePacketType{
+	"dropped": AttributePacketTypeDropped,
+	"errored": AttributePacketTypeErrored,
+	"success": AttributePacketTypeSuccess,
+}
+
+type metricNsxtInterfacePacketCount struct {
+	data     pmetric.Metric // data buffer for generated metric.
+	settings MetricSettings // metric settings provided by user.
+	capacity int            // max observed number of data points added to the metric.
+}
+
+// init fills nsxt.interface.packet.count metric with initial data.
+func (m *metricNsxtInterfacePacketCount) init() {
+	m.data.SetName("nsxt.interface.packet.count")
+	m.data.SetDescription("The number of packets flowing through the network interface on the node.")
+	m.data.SetUnit("{packets}")
+	m.data.SetDataType(pmetric.MetricDataTypeSum)
+	m.data.Sum().SetIsMonotonic(true)
+	m.data.Sum().SetAggregationTemporality(pmetric.MetricAggregationTemporalityCumulative)
+	m.data.Sum().DataPoints().EnsureCapacity(m.capacity)
+}
+
+func (m *metricNsxtInterfacePacketCount) recordDataPoint(start pcommon.Timestamp, ts pcommon.Timestamp, val int64, directionAttributeValue string, packetTypeAttributeValue string) {
+	if !m.settings.Enabled {
+		return
+	}
+	dp := m.data.Sum().DataPoints().AppendEmpty()
+	dp.SetStartTimestamp(start)
+	dp.SetTimestamp(ts)
+	dp.SetIntVal(val)
+	dp.Attributes().Insert(A.Direction, pcommon.NewValueString(directionAttributeValue))
+	dp.Attributes().Insert(A.PacketType, pcommon.NewValueString(packetTypeAttributeValue))
+}
+
+// updateCapacity saves max length of data point slices that will be used for the slice capacity.
+func (m *metricNsxtInterfacePacketCount) updateCapacity() {
+	if m.data.Sum().DataPoints().Len() > m.capacity {
+		m.capacity = m.data.Sum().DataPoints().Len()
+	}
+}
+
+// emit appends recorded metric data to a metrics slice and prepares it for recording another set of data points.
+func (m *metricNsxtInterfacePacketCount) emit(metrics pmetric.MetricSlice) {
+	if m.settings.Enabled && m.data.Sum().DataPoints().Len() > 0 {
+		m.updateCapacity()
+		m.data.MoveTo(metrics.AppendEmpty())
+		m.init()
+	}
+}
+
+func newMetricNsxtInterfacePacketCount(settings MetricSettings) metricNsxtInterfacePacketCount {
+	m := metricNsxtInterfacePacketCount{settings: settings}
+	if settings.Enabled {
+		m.data = pmetric.NewMetric()
+		m.init()
+	}
+	return m
+}
+
+type metricNsxtInterfaceThroughput struct {
+	data     pmetric.Metric // data buffer for generated metric.
+	settings MetricSettings // metric settings provided by user.
+	capacity int            // max observed number of data points added to the metric.
+}
+
+// init fills nsxt.interface.throughput metric with initial data.
+func (m *metricNsxtInterfaceThroughput) init() {
+	m.data.SetName("nsxt.interface.throughput")
```

Comment:
Should we use "network" instead of "interface" to match

https://github.com/open-telemetry/opentelemetry-specification/blob/7e35bec9401d9bc970a7107af45906c0890040b6/specification/metrics/semantic_conventions/system-metrics.md#systemfilesystem---filesystem-metrics